app/views/components/_related-navigation.html.erb

```diff
@@ -27,12 +28,30 @@
               <%= construct_section_heading(section_title) %>
             </h3>
           <% end %>
-          <nav role="navigation" class="app-c-related-navigation__nav-section" aria-labelledby="related-nav-<%= section_title %>">
+          <nav role="navigation" class="app-c-related-navigation__nav-section" aria-labelledby="related-nav-<%= section_title %>" data-module="toggle">
             <ul class="app-c-related-navigation_link-list" data-module="track-click">
-              <% links.each do |link| %>
-                <li class="app-c-related-navigation__link">
-                  <%= link_to link[:text], link[:path], class: "app-c-related-navigation__section-link" + construct_section_styling(section_title, defined_sections), rel: link[:rel] %>
-                </li>
+              <% constructed_link_array = [] %>
+              <% links.each.with_index(1) do |link, index| %>
+                <% if index <= max_section_length %>
+                  <li class="app-c-related-navigation__link">
+                    <%= link_to link[:text], link[:path], class: "app-c-related-navigation__section-link" + construct_section_styling(section_title, defined_sections), rel: link[:rel] %>
```

Comment:
Non-blocking nit: These line lengths are pretty long so are hard to read when they wrap, we could break them across multiple lines like

```erb
link_to(
  param,
  param,
  param
)
```